File: algorithms/372. Super Pow/solution.cc
class Solution {
    const int MOD = 1337;
    int modpow(int a, int b) {
        int r = 1;
        for ( int i = 0; i < b; ++i ) {
            r = (r * a) % MOD;
        }
        return r;
    }
public:
    int superPow(int a, vector<int>& b) {
        int r = 1;
        a %= MOD;
        for ( int d : b ) {
            r = modpow(r, 10) * modpow(a, d) % MOD;
        }
        return r;
    }
};

// 54 / 54 test cases passed.
// Status: Accepted
// Runtime: 13 ms
// beats 45.45 %

File: algorithms/372. Super Pow/solution.py
class Solution(object):
    def superPow(self, a, b):
        """
        :type a: int
        :type b: List[int]
        :rtype: int
        """
        s = 1
        for d in b:
            s = pow(s, 10, 1337) * pow(a, d, 1337) % 1337
        return s

# 54 / 54 test cases passed.
# Status: Accepted
# Runtime: 72 ms
# beats 81.95 %
